Document consent-managed and full tracking script options (#858)
mautic/user-documentation: docs/configuration/tracking_script.rst

After installation and setup of the :doc:`/configuration/cron_jobs` you're ready to begin tracking Contacts. You need to add a piece of JavaScript to the websites for each site you wish to track through Mautic.

This is a straightforward process. You can add this tracking script to your website template file, or install a Mautic Integration for the more common content management system platforms.

> To open 'Tracking Settings', click the settings wheel in the top right corner to open the Settings menu, then go to Configuration and select 'Tracking Settings'. See :doc:`/configuration/settings` for more about the Configuration area. Copy the exact code from 'Tracking Settings' in the Global Configuration and insert it just before the closing ``</body>`` tag. Mautic offers two setup options: a single all-in-one script, or two separate scripts for consent-managed setups. Mautic tracks Landing Pages automatically, so you only need these scripts on third-party websites.

Reviewer: ```suggestion
To open **Tracking Settings**:

#. Click the settings wheel in the top right corner to open the **Settings** menu
#. Go to **Configuration**
#. Select **Tracking Settings**.

See :doc:`/configuration/settings` for more information about the Configuration area. Copy the exact code from **Tracking Settings** in the Global Configuration and insert it just before the closing ``</body>`` tag. 

Mautic offers two setup options:

* A single all-in-one script
* Two separate scripts for consent-managed setups.

Mautic tracks Landing Pages automatically, so you only need these scripts on third-party websites.
```

.. vale off

Choosing a tracking setup
*************************

.. vale on

> 'Tracking Settings' presents both options as tabs. Pick the one that matches how your site handles consent:

Reviewer: ```suggestion
**Tracking Settings** presents both options as tabs. Pick the one that matches how your site handles consent:
```

* 'Full tracking'—use this when there's no separate consent gate, or when you handle consent elsewhere. It loads Contact tracking immediately.

> * 'Consent-managed'—use this when a consent manager gates tracking and cookies. Dynamic Content and Forms work for everyone, while Contact tracking waits until the visitor consents.

Reviewer: ```suggestion
* **Full tracking**: use this when there's no separate consent gate, or when you handle consent elsewhere. It loads Contact tracking immediately.
* **Consent-managed**: use this when a consent manager gates tracking and cookies. Dynamic Content and Forms work for everyone, while Contact tracking waits until the visitor consents.
```

.. vale off

Full tracking
*************

.. vale on

> The 'Full tracking' tab gives you a single all-in-one script. It loads the essential runtime first, then the tracking add-on, so use it only after the visitor grants tracking consent or where consent isn't required. This behaves like all-in-one tracking has always worked.

Reviewer: There are a couple of things:

- Use active instead of passive voice. See: 'isn't required'.
- Bold 'Full tracking' in the Full tracking tab.

> Here's an example of the all-in-one tracking JavaScript. Copy the whole block from 'Tracking Settings' in your Mautic instance rather than retyping it—the copied snippet already contains your instance's Site URL. In the example below, the placeholder ``example.com/mautic`` stands in for that URL:

Reviewer: ```suggestion
Here's an example of the all-in-one tracking JavaScript. Copy the whole block from **Tracking Settings** in your Mautic instance rather than retyping it. The copied snippet already contains your instance's Site URL. In the example below, the placeholder ``example.com/mautic`` stands in for that URL:
```

After installation and setup of the :doc:`/configuration/cron_jobs` you're ready to begin tracking Contacts. You need to add a piece of JavaScript to the websites for each site you wish to track through Mautic.

This is a straightforward process. You can add this tracking script to your website template file, or install a Mautic Integration for the more common content management system platforms.

To open **Tracking Settings**:

#. Click the settings wheel in the top right corner to open the **Settings** menu
#. Go to **Configuration**
#. Select **Tracking Settings**.

> See :doc:`/configuration/settings` for more information about the Configuration area. Copy the exact code from **Tracking Settings** in the Global Configuration and insert it just before the closing ``</body>`` tag.

Reviewer 2: This instruction also appears to cover the tracking add-on, which would enable tracking as soon as the essential script becomes ready. The merged UI explicitly says to run that snippet from the consent manager's callback after consent (`app/bundles/PageBundle/Translations/en_US/messages.ini:216-221`). Please scope the `</body>` instruction to the Full tracking and essential snippets, and direct users to run the add-on only from the callback.

Mautic offers two setup options:

* A single all-in-one script
* Two separate scripts for consent-managed setups.

Mautic tracks Landing Pages automatically, so you only need these scripts on third-party websites.

.. vale off

Choosing a tracking setup
*************************

.. vale on

**Tracking Settings** presents both options as tabs. Pick the one that matches how your site handles consent:

* **Full tracking**: use this when there's no separate consent gate, or when you handle consent elsewhere. It loads Contact tracking immediately.
* **Consent-managed**: use this when a consent manager gates tracking and cookies. Dynamic Content and Forms work for everyone, while Contact tracking waits until the visitor consents.

.. vale off

Full tracking
*************

.. vale on

The **Full tracking** tab gives you a single all-in-one script. It loads the essential runtime first, then the tracking add-on, so use it only after the visitor grants tracking consent or where your site doesn't require consent. This behaves like all-in-one tracking has always worked.

> Here's an example of the all-in-one tracking JavaScript. Copy the whole block from **Tracking Settings** in your Mautic instance rather than retyping it. The copied snippet already contains your instance's Site URL. In the example below, the placeholder ``example.com/mautic`` stands in for that URL:

Reviewer 2: This example is the legacy `/mtc.js` embed, not the code provided by the new **Full tracking** tab. That tab's loader sequentially loads `/mautic-essential.js` and `/mautic-tracking.js` (`app/bundles/PageBundle/Resources/views/FormTheme/Config/_config_trackingconfig_widget.html.twig:71-85`). Although `/mtc.js` remains compatible, readers cannot copy this example from the tab as stated. Please show the current loader or omit the example and retain only the backward-compatibility note.

    })(window,document,'script','https://example.com/mautic/mtc.js','mt');
   mt('send', 'pageview');

Existing ``/mtc.js`` embeds keep working unchanged, so you don't need to update sites that already use them.

.. vale off

Consent-managed tracking
************************

.. vale on

> The 'Consent-managed' tab gives you two snippets that you use together, and the loading order matters. The essential script loads first and runs safely before the visitor consents. The tracking add-on runs later, only after the visitor grants consent.

Reviewer: ```suggestion
The **Consent-managed** tab gives you two snippets that you use together, and the loading order matters. The essential script loads first and runs safely before the visitor consents. The tracking add-on runs later, only after the visitor grants consent.
```

.. note::

>    Separately embedded Focus Items track views and interactions independently of Mautic website tracking. Loading only the essential script doesn't turn off Focus Item tracking, so you still need to manage consent separately for those Focus Items. See :doc:`/channels/focus_items` for how to embed a Focus Item on your website.

Reviewer: Address the Vale warning:

- [Mautic.FeatureList] Is this referring to a Mautic feature? If so, use 'Channels' instead of 'channels'.

Essential script
================

> The essential script loads Mautic's essential features without visitor tracking. It renders Dynamic Content and injects Forms, but it doesn't track Contacts, set cookies, store data in the browser, or send identifying requests. That makes it safe to place on the page before the visitor consents.

Reviewer: Address the Vale warning:

- [Mautic.FeatureList] Is this referring to a Mautic feature? If so, use 'Landing Page' instead of 'page'.

> Copy the exact snippet from the 'Essential script (before consent)' area of the 'Consent-managed' tab. It powers :doc:`/components/dynamic_web_content` and :doc:`/components/forms` for every visitor.

Reviewer: ```suggestion
.. vale off

Copy the exact snippet from the **Essential script (before consent)** area of the **Consent-managed** tab. It powers :doc:`/components/dynamic_web_content` and :doc:`/components/forms` for every visitor.

.. vale on
```

Tracking add-on script
======================

> The tracking add-on adds the identity and tracking layer: Contact tracking, page-view events, and analytics Integrations such as Google Analytics and Facebook Pixel. You run this snippet from your consent manager's callback, after the visitor consents. The add-on is self-deferring: it runs immediately if the essential runtime is already ready, and otherwise waits for the essential-ready signal on its own before running and queuing the initial page view. You don't have to time this manually. This handoff is typically configured inside your consent or cookie-management platform, or handled by a developer, so if you're a non-developer this is the step to hand off along with the exact snippet.

Reviewer: ```suggestion
.. vale off

The tracking add-on adds the identity and tracking layer:

* Contact tracking
* Page-view events
* Analytics Integrations such as Google Analytics and Facebook Pixel.

You run this snippet from your consent manager's callback, after the visitor consents. The add-on is self-deferring. It runs immediately if the essential runtime is already ready, and otherwise waits for the essential-ready signal on its own before running and queuing the initial page view. You don't have to time this manually. This handoff is typically configured inside your consent or cookie-management platform, or handled by a developer, so if you're a non-developer, this is the step to hand off along with the exact snippet.

.. vale on
```

.. warning::

   Only place the tracking add-on when you also load the essential script. If you place the add-on without the essential script, it has nothing to wait for, never runs, and Mautic tracks nothing. Always load the essential script first.

> Copy the exact snippet from the 'Tracking add-on (after consent)' area of the 'Consent-managed' tab.

Reviewer: ```suggestion
.. vale off

Copy the exact snippet from the **Tracking add-on (after consent)** area of the **Consent-managed** tab.

.. vale on
```

    })(window,document,'script','https://example.com/mautic/mtc.js','mt');
   mt('send', 'pageview');

Existing ``/mtc.js`` embeds keep working unchanged, so you don't need to update sites that already use them.

.. vale off

Consent-managed tracking
************************

.. vale on

The **Consent-managed** tab gives you two snippets that you use together, and the loading order matters. The essential script loads first and runs safely before the visitor consents. The tracking add-on runs later, only after the visitor grants consent.

.. note::

   .. vale off

   Separately embedded Focus Items track views and interactions independently of Mautic website tracking. Loading only the essential script doesn't turn off Focus Item tracking, so you still need to manage consent separately for those Focus Items. See :doc:`/channels/focus_items` for how to embed a Focus Item on your website.

   .. vale on

Essential script
================

.. vale off

> The essential script loads Mautic's essential features without visitor tracking. It renders Dynamic Content and injects Forms, but it doesn't track Contacts, set cookies, store data in the browser, or send identifying requests. That makes it safe to place on the page before the visitor consents.

Reviewer 2: This overstates what the essential script provides. Before tracking, the implementation keeps existing DWC fallback content visible and makes no DWC request; personalized replacement begins only after both scripts load (`tests/acceptance/MauticScriptSplitCest.php:190-221,292-323`). It initializes Forms specifically inside DWC fallback slots (`:224-260`), while regular Form embeds work independently of tracking scripts (`app/bundles/FormBundle/Tests/Controller/PublicControllerFunctionalTest.php:18-38`). Please describe fallback content and fallback-slot Forms rather than Dynamic Content and Forms generally.
